src/UI/Component/Trigger/Triggerer.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace ILIAS\UI\Component\Trigger;
+
+/**
+ * Interface Triggerer
+ *
+ * Any component that can trigger other components must implement this interface
+ */
+interface Triggerer
+{
+
+    /**
+     * Trigger an action of another component
+     *
+     * @param TriggerAction $action
+     * @param string $event
+     * @return self
+     */
+    public function triggerAction(TriggerAction $action, $event = 'click');
```

Comment:
I think we need to allow multiple actions to be triggered. For example, a button might show a tooltip on hover and a modal on click. You're right, withTriggerAction would be consistent with the current syntax. But currently, the with* methods always overwrite a member (hope I'm correct) whereas in this case, we would stack the actions.